src/jit/lower.cpp

```diff
@@ -1175,12 +1184,29 @@ void Lowering::LowerArg(GenTreeCall* call, GenTreePtr* ppArg)
     {
 
 #ifdef _TARGET_ARMARCH_
+        // We need to handle double register arguments specially on SoftFP
+        // (secondReg is only for `double` type argument on SoftFP)
+        bool      isSoftFPDoubleRegArg = false;
+        regNumber secondReg            = (info->regNum != REG_STK) ?  REG_NEXT(info->regNum) : REG_STK;
+
         // For vararg call, reg args should be all integer.
         // Insert a copy to move float value to integer register.
         if ((call->IsVarargs() || comp->opts.compUseSoftFP) && varTypeIsFloating(type))
         {
-            var_types  intType = (type == TYP_DOUBLE) ? TYP_LONG : TYP_INT;
-            GenTreePtr intArg  = comp->gtNewOperNode(GT_COPY, intType, arg);
+            isSoftFPDoubleRegArg = (comp->opts.compUseSoftFP && type == TYP_DOUBLE && info->regNum != REG_STK);
+            var_types  intType   = (type == TYP_DOUBLE) ? TYP_LONG : TYP_INT;
+
+#ifdef _TARGET_ARM_
+            // We need gtOtherRegs[] for the case - copy from a double register to two int registers
+            GenTreePtr intArg = new (comp, GT_COPY) GenTreeCopyOrReload(GT_COPY, intType, arg);
```

Comment:
It looks like with FEATURE_MULTIREG_RET, there is an additional field. In gentree.h,

```cpp
4923 struct GenTreeCopyOrReload : public GenTreeUnOp
4924 {
4925 #if FEATURE_MULTIREG_RET
4926     // State required to support copy/reload of a multi-reg call node.
4927     // First register is is always given by gtRegNum.
4928     //
4929     regNumberSmall gtOtherRegs[MAX_RET_REG_COUNT - 1];
4930 #endif
```

I wanted to use `gtOtherRegs[0]` as the second register when we copy double register.